DSpace/DSpace
# Fixed community's sub-communities and collections sort

## References
* Fixes DSpace/DSpace-angular#2160
* Related DSpace/DSpace-angular#2239

## Description
Fixed community's sub-communities and collections endpoint sort options being hardcoded to sort by `dc.title,ASC`.

## Instructions for Reviewers
By default both endpoints still return their results sorted on the `dc.title` field in ascending order, but when the sort option is provided that order will be used instead.

**Guidance for how to test this PR:**
- Create a new community
- To test the `CommunityCollectionLinkRepository` backend:
  - Create 2 new collections with two different names in this new parent community
  - Go to `/core/communities/{uuid}/collections?sort=dc.title,ASC`
  - Check that the collections are sorted in ascending order
  - Go to `/core/communities/{uuid}/collections?sort=dc.title,DESC`
  - Check that the collections are sorted in descending order
- To test the `CommunitySubcommunityLinkRepository` backend:
  - Create 2 new sub communities with two different names in this new parent community
  - Go to `/core/communities/{uuid}/subcommunities?sort=dc.title,ASC`
  - Check that the sub communities are sorted in ascending order
  - Go to `/core/communities/{uuid}/subcommunities?sort=dc.title,DESC`
  - Check that the sub communities are sorted in descending order

## Checklist
